README.md

```diff
@@ -72,252 +84,351 @@ pull(
-and `blobId` respectively.
+The text inside a message can refer to three types of ssb-db entities: messages, feeds, and blobs (i.e. 
+attachments). Messages and blobs are referred to by their hashes, but a feed is referred to by its signing 
+public key. Thus, a message within a feed can refer to another feed, or to a particular point _within_ a feed.
 
-Note that `ssb-db` does not include facilities for retrieving a blob given the
-hash.
+Object ids begin with a sigil `@` `%` and `&` for a `feedId`, `msgId` and `blobId` respectively.
 
-### Replication
+Note that `ssb-db` does not include facilities for retrieving a blob given the hash.
 
-<link to ssb-replicate and ssb-ebt>
+## Replication
 
 It is possible to easily replicate data between two instances of `ssb-db`.
-First, they exchange maps of their newest data. Then, each one downloads
-all data newer than its newest data.
+First, they exchange maps of their newest data. Then, each one downloads all data newer than its newest data.
 
-[ssb-server](https://github.com/ssbc/ssb-server) is a tool that
-makes it easy to replicate multiple instances of ssb-db using a
-decentralized network.
+[ssb-server](https://github.com/ssbc/ssb-server) is a tool that makes it easy to replicate multiple instances 
+of ssb-db using a decentralized network.
 
-### Security properties
+## Security properties
 
-`ssb-db` maintains useful security properties even when it is
-connected to a malicious ssb-db database. This makes it ideal
-as a store for peer-to-peer applications.
+`ssb-db` maintains useful security properties even when it is connected to a malicious ssb-db database. This 
+makes it ideal as a store for peer-to-peer applications.
 
-Imagine that we want to read from a feed for which we know the identity,
-but we're connected to a malicious ssb-db instance. As long as the malicious
-database does not have the private key:
+Imagine that we want to read from a feed for which we know the identity, but we're connected to a malicious 
+ssb-db instance. As long as the malicious database does not have the private key:
 
 - The malicious database cannot create a new feed with the same identifier
 - The malicious database cannot write new fake messages to the feed
 - The malicious database cannot reorder the messages in the feed
-- The malicious database cannot send us a new copy of the feed that omits
-  messages from the middle
+- The malicious database cannot send us a new copy of the feed that omits messages from the middle
 - The malicious database *can* refuse to send us the feed, or only send
   us the first *N* messages in the feed
 - Messages may optionally be encrypted. See `test/end-to-end.js`.
 
 
 ## API
-
-### SecretStack.use(require('ssb-db')) => SecretStackApp
+## require('ssb-db')
+```js 
+SecretStack.use(require('ssb-db')) => SecretStackApp
+```
 
 Adds `ssb-db` persistence to a [secret-stack](https://github.com/ssbc/secret-stack) setup.
```

Comment:
Could start by saying... 
"The design pattern of `ssb-db` is for it to act as a plugin within the `SecretStack` plugin framework. The main export provides the plugin, which extends the `SecretStack` app with this plugins functionality, and API."

Reply:
Changed in 2eb92d0
Its ok?

Reply:
yeah